### PR TITLE
Vet clinic database: database performance audit

### PR DESCRIPTION
**Pull request by porag-m06**
# !!! Working alone with no partners assigned !!!!

screencapture-mail-google-mail-u-3-2023-08-18-12_03_32

# Highlights
- Query 01: ``EXPLAIN ANALYZE SELECT COUNT(*) FROM visits WHERE animal_id = 4;``
- Query 02: ``EXPLAIN ANALYZE SELECT * FROM visits where vet_id = 2;``
- Query 03: ``EXPLAIN ANALYZE SELECT * FROM owners WHERE email = 'owner_18327@mail.com';``

## Before Optimization: 
image

## Optimization By Indexing
image

## After Optimization: 
image

### CODE REVIEW

**Review by usorfaitheloho — COMMENTED** [review]
Hi @porag-m06 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is a pair programming activity but doesn't include commits of your coding partner. Kindly ensure you work with your partner and submit for a review

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by porag-m06** [comment]
Hi,
Thank you for the remainder, however, I am working alone this week with no partner assigned to me ( with confirmation from student success to work alone).

So, that would only be me working alone on this project.

Thanks

File: schema.sql [hunk 1]
```diff
@@ -57,4 +57,12 @@ CREATE TABLE visits (
     date_of_visit DATE,
     FOREIGN KEY (animal_id) REFERENCES animals(id),
     FOREIGN KEY (vet_id) REFERENCES vets(id)
-);
\ No newline at end of file
+);
+
```
Comment (HAMMAS-SALEEM) [inline schema.sql:61]: - [ ] Kindly attach the screenshot of the email from microverse for work alone permit in the pull request summary so the next reviewer will be able to proceed review of your pull request. 😊

**Review by HAMMAS-SALEEM — COMMENTED** [review]
# Invalid Submission :red_circle:

invalid-submission

## Hey @porag-m06 :wave:

Good job on implementing some of the requirements but I cannot proceed review of your pull request because there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval. Still, we highly recommend to students that they should take these changes into consideration._

Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by usorfaitheloho — COMMENTED** [review]
Hi @porag-m06 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is a pair programming project and should have commits from your partner. Since you said you are working alone, Kindly attach the screenshot of the email from Microverse for the work alone permit in the pull request summary so the next reviewer will be able to proceed review of your pull request. 😊

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: schema.sql (hunk 1 above)
Comment (porag-m06) [reply schema.sql:61]: screencapture-mail-google-mail-u-3-2023-08-18-12_03_32

**Comment by porag-m06** [comment]
screencapture-mail-google-mail-u-3-2023-08-18-12_03_32

**Review by danmainah — APPROVED** [review]
Hi @porag-m06 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ Added the extra data and run the queries and screenshot of execution time and added it in the pull request summary as requested
✔️  Added index to optimize query speed, providing a screenshot of the reduced execution time in the pull request summary as requested

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**